### PR TITLE
simplify "Compiling" info message: display relative path

### PR DESCRIPTION
**Pull request by hboutemy**
simplify current info message

```
[INFO] Compiling 29 source files to /full/path/to/project/which/is/very/noisy/target/test-classes
```

by replacing full path with relative to basedir:

```
[INFO] Compiling 29 source files to target/test-classes
```

### CODE REVIEW

**Review by olamy — APPROVED** [review]

File: plexus-compilers/plexus-compiler-javac/src/main/java/org/codehaus/plexus/compiler/javac/JavacCompiler.java [hunk 1]
```diff
@@ -143,9 +143,11 @@ public CompilerResult performCompile( CompilerConfiguration config )
 
         if ( ( getLogger() != null ) && getLogger().isInfoEnabled() )
         {
+            String to = ( config.getWorkingDirectory() == null ) ? config.getOutputLocation() :
+                config.getWorkingDirectory().toPath().relativize( destinationDir.toPath() ).toString();
             getLogger().info( "Compiling " + sourceFiles.length + " " +
                                   "source file" + ( sourceFiles.length == 1 ? "" : "s" ) +
-                                  " to " + destinationDir.getAbsolutePath() );
+                                  " to " + to );
```
Comment (pzygielo) [inline plexus-compilers/plexus-compiler-javac/src/main/java/org/codehaus/plexus/compiler/javac/JavacCompiler.java:150]: Could these duplications be avoided?

Comment (olamy) [reply plexus-compilers/plexus-compiler-javac/src/main/java/org/codehaus/plexus/compiler/javac/JavacCompiler.java:150]: > Could these duplications be avoided?

?? not sure to understand your concern?

Comment (hboutemy) [reply plexus-compilers/plexus-compiler-javac/src/main/java/org/codehaus/plexus/compiler/javac/JavacCompiler.java:150]: you mean create a method in AbstractCompiler for logging, and have plexus compilers use this method?

https://codehaus-plexus.github.io/plexus-compiler/plexus-compiler-api/apidocs/org/codehaus/plexus/compiler/AbstractCompiler.html

Comment (pzygielo) [reply plexus-compilers/plexus-compiler-javac/src/main/java/org/codehaus/plexus/compiler/javac/JavacCompiler.java:150]: `to` and message logged/being output are calculated twice (Javac, AspecJCompiler) in the same way.

Comment (hboutemy) [reply plexus-compilers/plexus-compiler-javac/src/main/java/org/codehaus/plexus/compiler/javac/JavacCompiler.java]: good idea, done in 37f6843f4dfbf0a8a9caa185926d16ab8b55f0e8

Comment (pzygielo) [reply plexus-compilers/plexus-compiler-javac/src/main/java/org/codehaus/plexus/compiler/javac/JavacCompiler.java]: Great!

**Pull request merged**